lib/phoenix/endpoint/supervisor.ex

```diff
@@ -11,17 +11,47 @@ defmodule Phoenix.Endpoint.Supervisor do
   Starts the endpoint supervision tree.
   """
   def start_link(otp_app, mod, opts \\ []) do
-    case Supervisor.start_link(__MODULE__, {otp_app, mod, opts}, name: mod) do
-      {:ok, _} = ok ->
-        warmup(mod)
-        log_access_info(otp_app, mod)
-        ok
-
+    with {:ok, _} <- check_force_ssl(otp_app, mod),
+         {:ok, _} = ok <- Supervisor.start_link(__MODULE__, {otp_app, mod, opts}, name: mod) do
+      warmup(mod)
+      log_access_info(otp_app, mod)
+      ok
+    else
       {:error, _} = error ->
         error
     end
   end
 
+  defp check_force_ssl(otp_app, mod) do
+    compile_config = mod.__compile_config__() |> Keyword.get(:force_ssl)
+
+    runtime_config =
+      config(otp_app, mod)
```

Comment:
I don't think we should build the config list again, as we already do on init. My suggestion is to move this logic to inside the `init` function.

Reply:
Oh good catch. Moved.